src/modules/fancyzones/lib/ZoneWindow.cpp

```diff
@@ -148,7 +148,7 @@ namespace ZoneWindowDrawUtils
                            COLORREF highlightColor,
                            int zoneOpacity,
                            const std::vector<winrt::com_ptr<IZone>>& zones,
-                           const winrt::com_ptr<IZone>& highlightZone,
+                           const std::vector<int>& highlightZone,

```

Comment:
Zones vector is not protected against async calls, it could be cleared in other method. 
I had a crash at `isHighlighted[x] = true;` since `x` was out of range.

Reply:
I discussed this with @yuyoyuppe, we think we should add a mutex to ZoneWindow. We believe that this happens because both ZoneWindow and FancyZones have their own WndProc, which run in separate threads, and both have access to ZoneWindow's data.

Reply:
so perhaps we should do it for this PR?

Reply:
It can be passed as a copy, 
also highlightZone -> highlightZones